### ISSUE TITLE
NVDA emits selection events when navigating with arrow keys in mark mode of windows terminal preview

### ISSUE THREAD

**Issue by beqabeqa473** [nvaccess/nvda#13874]
### Steps to reproduce:
Open windows terminal preview;
Press ctrl+shift+m;
Navigate with arrow keys
### Actual behavior:
NVDA announces selected current char, unselected previous char on every single character.
> Press right arrow
> NVDA announces `unselected a, selected b`
### Expected behavior:
NVDA should read just characters when navigating with arrow keys without modifiers.
> Press right arrow
> NVDA announces `a`
#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
alpha-25756,ba743746
#### Windows version:
Windows 10 build 19044.1766
#### Name and version of other software in use when reproducing the issue:
Windows terminal preview
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

**Comment by beqabeqa473** [comment]
Filled the issues gainst Terminal.
https://github.com/microsoft/terminal/issues/13447

**Comment by seanbudd** [comment]
Is this a new issue? Can you reproduce it with 2022.2beta3?
cc @codeofdusk

**Comment by beqabeqa473** [comment]
Yes.

It is a new feature of a windows terminal

**Comment by seanbudd** [comment]
@beqabeqa473 - we recently changed the API we use for Windows Terminal in NVDA alpha. Can you confirm if this is still a problem when using 2022.2beta3? 

You can also confirm this via changing settings:
>By changing the "Windows Console support" setting in NVDA's advanced settings panel.

We've fixed up your issue description, can you confirm it is accurate / fix it up further.

**Comment by beqabeqa473** [comment]
hello.

yes, i've tried a 2022.2 beta 3 now and issue persists.
also, i changed settings related to terminal in advanced settings and behaviour stays the same.

**Comment by seanbudd** [comment]
Thanks

**Comment by codeofdusk** [comment]
> we recently changed the API we use for Windows Terminal

No, we haven't. Windows Terminal is (and has been) *always* UIA, because the legacy APIs are unavailable there.

I haven't investigated this but strongly suspect it's a wt bug/needs a fix on their side. CC @carlos-zamora.

**Comment by beqabeqa473** [comment]
Are there any news in this area?

Latest terminal 1.17 doesn't fix this issue.

**Comment by Adriani90** [comment]
@beqabeqa473 are you still having this issue in the last terminal version?

**Comment by beqabeqa473** [comment]
Very easy to test.
Yes, of course, unfortunately problems are not dissappearing magically

**Comment by Adriani90** [comment]
Note that in Powershell 7 this issue does not occur.

**Comment by Adriani90** [comment]
It doesn't also occur with Windows powershell 5.1.
It seems it really occurs only with the Windows terminal app. Tested with Terminal 1.21.3231.0.

**Comment by seanbudd** [comment]
Is this still an issue with the latest terminal?

**Comment by devinprater** [comment]
On the newest Windows terminal, Windows 11, and NVDA Version: 2025.3rc1 (2025.3.0.52513) it is fixed for me.

**Comment by seanbudd** [comment]
Closing as fixed, thanks @devinprater
